packages/app-mobile/root.tsx

```diff
@@ -207,7 +207,13 @@ const generalMiddleware = (store: any) => (next: any) => async (action: any) =>
 		setTimeLocale(Setting.value('locale'));
 	}
 
-	if ((action.type === 'SETTING_UPDATE_ONE' && (action.key.indexOf('encryption.') === 0)) || (action.type === 'SETTING_UPDATE_ALL')) {
+	// Like the desktop and CLI apps, we run this whenever the sync target properties change.
+	// Previously, this only ran when encryption was enabled/disabled. However, after fetching
+	// a new key, this needs to run and so we run it when the sync target info changes.
+	if (
+		(action.type === 'SETTING_UPDATE_ONE' && ['encryption.passwordCache', 'encryption.masterPassword', 'syncInfoCache'].includes(action.key))
+		|| action.type === 'SETTING_UPDATE_ALL'
+	) {
```

Comment:
Compare with the equivalent reducer logic for the desktop/CLI apps:

https://github.com/laurent22/joplin/blob/ac2258769a79aa1d9b5d5214918ac02c16579b9b/packages/lib/BaseApplication.ts#L393
https://github.com/laurent22/joplin/blob/ac2258769a79aa1d9b5d5214918ac02c16579b9b/packages/lib/BaseApplication.ts#L419